capellambse/_diagram_cache.py

```diff
@@ -219,6 +220,255 @@ def _sanitize_filename(fname: str) -> str:
+      first point.
+
+    For the purpose of calculating the extents (i.e., the minimum and
+    maximum x and y coordinates), the closure of the shape does not
+    affect the calculation. You simply need to find the smallest and
+    largest x and y values among all the points. Therefore, the logic
+    for calculating the extents of a polyline can be directly applied to
+    a polygon.
+
+    Parameters
+    ----------
+    element : etree.Element
+        The XML element representing the polygon.
+
+    Returns
+    -------
+    tuple[float, float, float, float]
+        (x_min, x_max, y_min, y_max)
+    """
+    return polyline_extents(element)
+
+
+def polyline_extents(
+    element: etree.Element,
+) -> tuple[float, float, float, float]:
+    """Compute extents for a polyline.
+
+    Parameters
+    ----------
+    element : etree.Element
+        The XML element representing the polyline.
+
+    Returns
+    -------
+    tuple[float, float, float, float]
+        (x_min, x_max, y_min, y_max)
+    """
+    points = element.get("points", "")
+    points = [tuple(map(float, p.split(","))) for p in points.strip().split()]
+    x_coords, y_coords = zip(*points)
+    return (min(x_coords), max(x_coords), min(y_coords), max(y_coords))
+
+
+def rect_extents(element: etree.Element) -> tuple[float, float, float, float]:
+    """Compute extents for a rectangle.
+
+    Parameters
+    ----------
+    element : etree.Element
+        The XML element representing the rectangle.
+
+    Returns
+    -------
+    tuple[float, float, float, float]
+        (x_min, x_max, y_min, y_max)
+    """
+    width_str = element.get("width", "0")
+    height_str = element.get("height", "0")
+    try:
+        width = float(width_str)
+        height = float(height_str)
+    except ValueError:
+        return -1.0, -1.0, -1.0, -1.0
+    x = float(element.get("x", 0))
+    y = float(element.get("y", 0))
+    width = float(element.get("width", 0))
+    height = float(element.get("height", 0))
+    return (x, x + width, y, y + height)
+
+
+def text_extents(element: etree.Element) -> tuple[float, float, float, float]:
+    """Compute extents for text.
+
+    Simplified assumption of width based on character count and font
+    size.
+
+    Parameters
+    ----------
+    element : etree.Element
+        The XML element representing the text.
+
+    Returns
+    -------
+    tuple[float, float, float, float]
+        (x_min, x_max, y_min, y_max)
+    """
+    x = float(element.get("x", 0))
+    y = float(element.get("y", 0))
+    text = element.text or ""
+    font_size = float(element.get("font-size", 16))  # Default font size
+    width = len(text) * font_size * 0.6  # Simplistic width estimation
+    return (x, x + width, y - font_size, y)
+
+
+def _svg_content_bbox(
+    root: lxml.etree.Element,
+) -> tuple[float, float, float, float]:
+    x_min = y_min = float("inf")
+    x_max = y_max = float("-inf")
+    funcs = set()
```

Comment:
The `funcs` set exists only for debug/print usage. We can remove this object and log the requested `func_name` right when it is done.

Reply:
This would lead to significantly more log messages though. The set only retains one call for each function, while this approach would emit one log message per call, producing potentially hundreds of messages for a busy diagram.

I don't see the need to keep this in at all, the final algorithm here is so simple that we don't really need these debug messages anymore. Let's just get rid of them entirely. (Also including the "unknown shape extender" error you suggested in the next comment - there's just way too many elements we intentionally don't handle.)